homeassistant/components/synology_dsm/config_flow.py

```diff
@@ -262,14 +247,27 @@ async def async_step_ssdp(self, discovery_info: DiscoveryInfoType) -> FlowResult
         self.context["title_placeholders"] = self.discovered_conf
         return await self.async_step_user()
 
-    async def async_step_reauth(
+    async def async_step_reauth(self, data: dict[str, Any]) -> FlowResult:
+        """Perform reauth upon an API authentication error."""
+        self.reauth_conf = data.copy()
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
         self, user_input: dict[str, Any] | None = None
     ) -> FlowResult:
-        """Perform reauth upon an API authentication error."""
-        self.reauth_conf = self.context.get("data", {})
-        self.reauth_reason = self.context.get(EXCEPTION_DETAILS)
-        if user_input is None:
-            return await self.async_step_user()
+        """Perform reauth confirm upon an API authentication error."""
+        if not user_input:
+            return self.async_show_form(
+                step_id="reauth_confirm", data_schema=_reauth_schema()
+            )
+
+        self.reauth_conf.update(
+            {
+                CONF_USERNAME: user_input[CONF_USERNAME],
+                CONF_PASSWORD: user_input[CONF_PASSWORD],
+            }
+        )
+        user_input.update(self.reauth_conf)
         return await self.async_step_user(user_input)
```

Comment:
```suggestion
        return await self.async_step_user({**self.reauth_conf, **user_input})
```

Reply:
Downside with passing the input to the user step for validation is that we'll potentially show the full form if there's an error.

Reply:
ah ... great, thx 👍